File: CorejavaExample/MyOwnAutoShopProject/src/com/sonata/Car.java
package com.sonata;

public class Car
{
	int speed;
	double regularPrice;
	String color;
public Car() {}
	
	public Car(int a, double b, String c) {
		this.speed=a;
		this.regularPrice=b;
		this.color=c;
		getSalePrice();
	}
	public Car(int a, double b, String c,int d,int e) {
		this.speed=a;
		this.regularPrice=b;
		this.color=c;
		getSalePrice();
	}
	public Car(int a, double b, String c,int d) {
		this.speed=a;
		this.regularPrice=b;
		this.color=c;
		getSalePrice();
	}
	public double getSalePrice() {
		return regularPrice;
	}
	public void display() {
		System.out.println(speed);
		System.out.println(regularPrice);
		System.out.println(color);
		System.out.println(getSalePrice());
	
	}
public static void main(String[]arg) {
	Car c1=new Car(70,20000,"white");
	Sedan s1 = new Sedan(230, 550000,"black", 15);
	Ford f1 = new Ford (210,480000,"blue",2021, 10000);
	Truck t1 = new Truck(130,250000,"red",2500);
	c1.display();
	System.out.println("Salesprice of sedan: "+s1.getSalePrice());
	System.out.println("Saleprice of ford: "+f1.getSalePrice());
	System.out.println("Saleprice of Truck: "+t1.getSalePrice());
	
}

}

File: Typescript_2021/Car.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var Car_1 = require("./Car");
var Truck = /** @class */ (function (_super) {
    __extends(Truck, _super);
    function Truck() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Truck.prototype.getSalePrice = function () {
        if (this.weight > 2000) {
            var prc = this.regularPrice - (this.regularPrice * .1);
            console.log(prc);
        }
        else {
            var prc = this.regularPrice - (this.regularPrice * .2);
            console.log(prc);
        }
    };
    return Truck;
}(Car_1.Car));
var t1 = new Truck();
t1.weight = 2001;
t1.regularPrice = 2000;
t1.speed = 100;
t1.color = "Red";
t1.getSalePrice();
